stdlib/public/core/UnicodeSPI.swift

```diff
@@ -0,0 +1,125 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+//===----------------------------------------------------------------------===//
+// Unicode.NFD
+//===----------------------------------------------------------------------===//
+
+extension Unicode {
+  @_spi(_Unicode)
+  @available(SwiftStdlib 5.7, *)
+  public struct _NFD {
+    let base: Substring.UnicodeScalarView
```

Comment:
To avoid surprises, I recommend spelling out the memberwise initializer here. The compiler-provided one _probably_ works fine, but it usually is a good idea to be explicit about things that might get exposed.

(E.g. I don't know for sure if the implicit initializer will be declared public or internal. `internal` would make more sense, but ¯\\\_(ツ)\_/¯)